app/src/main/java/in/co/rajkumaar/amritarepo/timings/PublicTransportsActivity.java

```diff
@@ -245,119 +261,313 @@ public void onCancelled(@NonNull DatabaseError error) {
+                            flag=1;
+                            Date startTime=currentTime.getTime();
+                            Date endTime=trainTime.getTime();
+                            long timediff=endTime.getTime()-startTime.getTime();
+                            countdown(timediff);
+                        }
+                    }
+                    if(flag==0)
+                    {
+                        nextTrainBus.setText("No Trains");
+                    }
+                } catch (JSONException | ParseException e) {
+                    e.printStackTrace();
                 }
             }
         }
 
         if (type != null && type.equals("Trains to Palghat")) {
+
+            trainbus.setImageResource(R.drawable.trainimage);
+            flag=0;
             String json = preferences.getString("trains-to-pkd", null);
             if (json != null) {
                 JSONArray timings = new JSONArray(json);
                 System.out.println(timings.toString());
-                for (int i = 0; i < timings.length(); ++i) {
-                    JSONObject item = timings.getJSONObject(i);
-                    items.add(new DataItem(
-                            item.getString("name"),
-                            item.getString("days"),
-                            item.getString("dep"),
-                            "etmd",
-                            "pkd",
-                            "train"
-                    ));
+                try {
+                    Calendar currentTime = Calendar.getInstance();
+                    Log.e("TAG", "loadData: " + currentTime.getTime());
+                    for (int i = 0; i < timings.length(); ++i) {
+                        JSONObject item = timings.getJSONObject(i);
+                        Date busTimeDate = dateFormat.parse(item.getString("dep"));
+                        Calendar trainTime = Calendar.getInstance();
+                        trainTime.setTime(new Date());
+                        trainTime.set(Calendar.HOUR_OF_DAY, busTimeDate.getHours());
+                        trainTime.set(Calendar.MINUTE, busTimeDate.getMinutes());
+                        Log.e("TAG", "loadData: " + trainTime.getTime());
+                        items.add(new DataItem(
+                                item.getString("name"),
+                                item.getString("days"),
+                                item.getString("dep"),
+                                "etmd",
+                                "pkd",
+                                "train"
+                        ));
+                        if(currentTime.get(Calendar.DAY_OF_WEEK)==1 && item.getString("days").equals("All Days Except Sundays"))
+                        {
+                            continue;
+                        }
+                        else if(currentTime.get(Calendar.DAY_OF_WEEK)==5 && item.getString("days").equals("All Days Except Thursdays"))
```

Comment:
Can you write a more generic logic for this ? Like get current `DAY` in string as for eg : `Friday` and check if `item.getString("days").toLowerCase().contains(currentDay.toLowerCase())`?